addons/website_event/controllers/main.py

```diff
@@ -215,22 +217,22 @@ def _prepare_event_register_values(self, event, **post):
         }
 
     @http.route('/event/add_event', type='json', auth="user", methods=['POST'], website=True)
-    def add_event(self, event_name="New Event", **kwargs):
-        event = self._add_event(event_name, request.context)
-        return "/event/%s/register?enable_editor=1" % slug(event)
-
-    def _add_event(self, event_name=None, context=None, **kwargs):
-        if not event_name:
-            event_name = _("New Event")
-        date_begin = datetime.today() + timedelta(days=(14))
-        vals = {
+    def add_event(self, event_name, event_start, event_end, address_id, **kwargs):
+        date_begin = pytz.timezone(request.env.user.tz).localize(parse(event_start)).astimezone(pytz.utc).replace(tzinfo=None)
+        date_end = pytz.timezone(request.env.user.tz).localize(parse(event_end)).astimezone(pytz.utc).replace(tzinfo=None)
+        values = {
             'name': event_name,
-            'date_begin': fields.Date.to_string(date_begin),
-            'date_end': fields.Date.to_string((date_begin + timedelta(days=(1)))),
+            'date_begin': date_begin,
+            'date_end': date_end,
+            'address_id': int(address_id) if address_id else request.env['res.partner'],
             'seats_available': 1000,
             'website_id': request.website.id,
         }
-        return request.env['event.event'].with_context(context or {}).create(vals)
+        event = self._add_event(values)
+        return "/event/%s/register?enable_editor=1" % slug(event)
+
+    def _add_event(self, values):
```

Comment:
I would have a _prepare_event_values function (like we do in other addons), that returns the values. And caller will simply create based on values. That way override may use the result of super() to have all available values before adding their own.